### DISCUSSION TITLE
First sight questions

### DISCUSSION THREAD

**Discussion by ambarusa** [Zer0-bit/gaggiuino#205] [Q&A]
Hello. I feel like I'm ready to upgrade my Gaggia Pro, and I was super happy to see that there is already a project for PID controller, with a nice display, preset and layout. Actually I was planning to use an ESP chip to make it smart, turn it on from an app. 

I had couple of questions in my mind:
How is the system calibrated? Is there some auto-calibration feature like in the analog one?
Is it possible to use other thermocouple, if the one you specified is not available? (I think this is connected to the calibration)
How is the process when you are frothing milk?
How do you stand with the scaler?

Thank you so much in advance.

**Comment by anacruza** [comment]
I'm working (very slowly) in a "Gaggiesp32" project! I'm using Home Assistant and Esphome firmware in the ESP32. Esphome has an integrated PID controller with an autotune function that I'm about to try in my Gaggia. In the bench it seems to work ok.

**Reply by ambarusa** [reply]
Thanks. What thermocouple do you use?

**Reply by anacruza** [reply]
I'm using this setting:

https://es.aliexpress.com/item/32835036293.html?gatewayAdapt=glo2esp&spm=a2g0o.order_list.0.0.1875194dM9UnW7
https://es.aliexpress.com/item/32841448771.html?gatewayAdapt=glo2esp&spm=a2g0o.order_list.0.0.1875194dM9UnW7
https://es.aliexpress.com/item/4000899938277.html?gatewayAdapt=glo2esp&spm=a2g0o.order_list.0.0.1875194dM9UnW7

**Reply by ambarusa** [reply]
Is there a reason you chose ESP32 in the first place?

**Reply by ambarusa** [reply]
Do you have any idea what is with the 2.1.4 Steam Config Chapter in this documentation? what is need to be done for the Steaming (other than setting new temperature)?

**Reply by anacruza** [reply]
> Is there a reason you chose ESP32 in the first place?

I'm currently using esphome in home assistant, and that way I can get the best of two worlds: having an internal logic programmed into the esp32 (via esphome sensors and lambdas) and the ability to control and monitor via WiFi (for example, home assistant phone app can use my morning alarm to trigger an automation that turns on the coffee machine on week days.

**Comment by ambarusa** [comment]
I'm also using Home Assistant, and this is why I want to add an additional relay at the first place, so the chip is always on, and I can turn the machine on remotely. I just have 8266's unused at home, and I thought it may not be good enough.
